.github/copilot-instructions.md

```diff
@@ -0,0 +1,152 @@
+
+Always reference these instructions first and fallback to search or bash commands only when you encounter unexpected information that does not match the info here.
+
+## Working Effectively
+
+### Prerequisites and Dependencies
+- **Go 1.25+** is required (specified in go.mod)
+- **CGO is REQUIRED** - this project depends on C++ libraries 
+- **C++ compiler** (g++ or equivalent) is required for building
+- **Linux/amd64** is the primary supported platform
+- Dependencies: `libstdc++` and `libgcc` runtime libraries
+
+### Build and Test Commands
+- **CRITICAL**: NEVER CANCEL long-running commands. Set timeouts appropriately.
+
+#### Initial Setup
+```bash
+go version  # Verify Go 1.25+ is installed
+go mod download  # Takes ~25 seconds, NEVER CANCEL
+```
+
+#### Build the Project
+```bash
+go build -v ./...  # Takes ~50 seconds, NEVER CANCEL - set timeout to 90+ minutes for safety
+go build -o Zeno .  # Build main executable, takes ~3 seconds
+```
+
+#### Test the Project
+```bash
+# Setup test environment (required for e2e tests)
+mkdir -p /tmp/unit_coverage /tmp/e2e_coverage
+echo 0 | sudo tee /proc/sys/kernel/apparmor_restrict_unprivileged_userns
+
+# Unit tests - Takes ~85 seconds, NEVER CANCEL - set timeout to 30+ minutes
+go test -v -p=12 -race -cover -covermode=atomic -coverpkg=./... $(go list ./... | grep -v /e2e/test/) -args -test.gocoverdir=/tmp/unit_coverage
+
+# E2E tests - Takes ~47 seconds, NEVER CANCEL - set timeout to 40+ minutes  
+go test -v -p=12 -race -cover -covermode=atomic -coverpkg=./... $(go list ./e2e/test/...) -args -test.gocoverdir=/tmp/e2e_coverage
+```
+
+#### Code Quality
+```bash
+go vet ./...        # Fast linting check
+gofmt -l .          # Check formatting (some files may need formatting)
+```
+
+### Run the Application
+
+#### CLI Usage
+```bash
+./Zeno --help               # Show main help
+./Zeno get --help           # Show crawling options  
+./Zeno get url --help       # Show URL archiving help
+./Zeno get url https://example.com  # Archive a single URL
+```
+
+#### Available Commands
+- `Zeno get url [URL...]` - Archive given URLs to WARC files
+- `Zeno get hq` - Start crawling with crawl HQ connector
+- Use `--tui` flag for terminal user interface
+
+### Docker Support
+- **Docker build currently fails** in sandbox environments due to network restrictions
+- Dockerfile exists but requires internet access for Alpine package installation
+- Use local Go build instead of Docker for development
+
+## Validation and Testing
+
+### Manual Validation Steps
+After making changes, ALWAYS:
+1. **Build successfully**: `go build -o Zeno .`
+2. **Run help command**: `./Zeno --help` to verify CLI works
+3. **Check binary dependencies**: `ldd ./Zeno` to verify C++ linkage
+4. **Run formatting check**: `gofmt -l .` and fix any issues
+5. **Run vet**: `go vet ./...` to catch common issues
```

Comment:
1. Try to crawl example website
2. (if applicable) attempt to crawl using HQ as well